controllers/v1/questionsController.js

```diff
@@ -685,4 +685,586 @@ module.exports = class Questions extends Abstract {
+          pointerToQuestionData < questionData.length;
+          pointerToQuestionData++
+        ) {
+
+          let parsedQuestion = gen.utils.valueParser(
+            questionData[pointerToQuestionData]
+          );
+
+          if (
+            (parsedQuestion["hasAParentQuestion"] == "YES" &&
+              !questionCollection[parsedQuestion["parentQuestionId"]]) ||
+            (parsedQuestion["instanceParentQuestionId"] !== "NA" &&
+              !questionCollection[parsedQuestion["instanceParentQuestionId"]])
+          ) {
+            pendingItems.push({
+              parsedQuestion: parsedQuestion,
+              criteriaToBeSent: criteria,
+              evaluationFrameworkMethod: ecm,
+              section: section
+            });
+          } else {
+
+            let question = await questionInCsv(parsedQuestion);
+
+            let resultFromCreateQuestions = await createQuestion(
+              parsedQuestion,
+              question,
+              criteria,
+              ecm,
+              section
+            );
+
+            if (resultFromCreateQuestions.result) {
+              questionCollection[resultFromCreateQuestions.result.externalId] =
+                resultFromCreateQuestions.result;
+            }
+            input.push(resultFromCreateQuestions.total[0]);
+          }
+        }
+
+        if (pendingItems) {
+          for (
+            let pointerToPendingData = 0;
+            pointerToPendingData < pendingItems.length;
+            pointerToPendingData++
+          ) {
+
+            let eachPendingItem = pendingItems[pointerToPendingData];
+
+            let question = questionInCsv(eachPendingItem);
+
+            let csvQuestionData = await createQuestion(
+              eachPendingItem.parsedQuestion,
+              question,
+              eachPendingItem.criteriaToBeSent,
+              eachPendingItem.evaluationFrameworkMethod,
+              eachPendingItem.section
+            );
+
+            input.push(csvQuestionData.total[0]);
+          }
+        }
+
+        input.push(null);
+
+      } catch (error) {
+        reject({
+          message: error
+        });
+      }
+    });
+  }
+
+
+  /**
+   * @api {post} /assessment/api/v1/questions/bulkUpdateSurveyQuestions?solutionExternalId=:solutionExternalId Bulk Update Survey Questions CSV
+   * @apiVersion 1.0.0
+   * @apiName Bulk Update Survey Questions CSV
+   * @apiGroup Questions
+   * @apiParam {File} questions Mandatory questions file of type CSV.
+   * @apiUse successBody
+   * @apiUse errorBody
+   */
+
+  /**
+   * Survey Questions bulk update.
+   * @method
+   * @name bulkUpdateSurveyQuestions
+   * @param {Object} req - requested data.
+   * @param {Object} req.files.questions - bulk update questions csv 
+   * consisting of SYSTEM_ID.
+   * @returns {CSV} - same csv with extra field UPDATE_STATUS to indicate 
+   * whether the question is updated or not.  
+   */
+
+  bulkUpdateSurveyQuestions(req) {
```

Comment:
@deepakr123 - Suggest merging it with existing API.

Reply:
resolved